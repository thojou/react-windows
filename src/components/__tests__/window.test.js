import 'jest-styled-components';
import React from 'react';
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Window from "../window";

describe('WindowBody', () => {
    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element, node) => {
            return element;
        })
    });

    afterEach(() => {
        ReactDOM.createPortal.mockClear();
    });

    it('should render', () => {
        const tree = renderer.create(<Window size={{
            width: 200,
            height: 100
        }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});