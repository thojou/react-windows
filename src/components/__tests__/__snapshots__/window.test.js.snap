// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowBody should render 1`] = `
.c0 {
  overflow: hidden;
  height: 100%;
  box-shadow: 0 3px 6px #A7893D;
}

.c6 {
  display: block;
  height: 100%;
}

.c1 {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  display: grid;
  color: #666666;
  font: normal bold 16px/21px "Segoe UI";
  background-color: #ffffff;
  border: 1px solid #E8E8E8;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto 1fr;
  grid-template-areas: 'title action' 'body body';
}

.c5 {
  height: 100%;
  grid-area: body;
  box-sizing: border-box;
}

.c5 > * {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

.c2 {
  height: 100%;
  grid-area: title;
  box-sizing: border-box;
  padding: 0.5rem 0.938rem;
  background-color: #E8E8E8;
}

.c2 > * {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

.c3 {
  height: 100%;
  grid-area: action;
  box-sizing: border-box;
  padding: 0.5rem 0.938rem;
  background-color: #E8E8E8;
}

.c3 > * {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

.c4 {
  border: none;
  background-color: transparent;
  color: #666666;
  cursor: pointer;
}

.c4:hover {
  color: #A7893D;
}

.c4:last-child {
  padding-right: 0;
}

<div
  className="react-draggable"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "boxSizing": "border-box",
      "cursor": "move",
      "display": "inline-block",
      "flexShrink": 0,
      "height": "100px",
      "left": 0,
      "maxHeight": 9007199254740991,
      "maxWidth": 9007199254740991,
      "minHeight": undefined,
      "minWidth": undefined,
      "position": "absolute",
      "top": 0,
      "touchAction": "none",
      "transform": "translate(-100px,-50px)",
      "userSelect": "auto",
      "width": "200px",
    }
  }
  transform={null}
>
  <div
    className="c0"
  >
    <div
      className="c1"
      rows="auto 1fr"
    >
      <div
        className="c2 window-draggable"
        name="title"
      >
        WindowName
      </div>
      <div
        className="c3 window-draggable"
        name="action"
      >
        <div>
          <button
            aria-label="minimize"
            className="c4"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 352H48c-26.5 0-48 21.5-48 48v32c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-32c0-26.5-21.5-48-48-48z"
              />
            </svg>
          </button>
          <button
            aria-label="maximize"
            className="c4"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-16 160H64v-84c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12v84z"
              />
            </svg>
          </button>
          <button
            aria-label="close"
            className="c4"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="c5"
        name="body"
      >
        <div
          className="c6"
        />
      </div>
    </div>
  </div>
  <span>
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "row-resize",
          "height": "10px",
          "left": "0px",
          "position": "absolute",
          "top": "-5px",
          "userSelect": "none",
          "width": "100%",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "position": "absolute",
          "right": "-5px",
          "top": "0px",
          "userSelect": "none",
          "width": "10px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-5px",
          "cursor": "row-resize",
          "height": "10px",
          "left": "0px",
          "position": "absolute",
          "userSelect": "none",
          "width": "100%",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "left": "-5px",
          "position": "absolute",
          "top": "0px",
          "userSelect": "none",
          "width": "10px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "ne-resize",
          "height": "20px",
          "position": "absolute",
          "right": "-10px",
          "top": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-10px",
          "cursor": "se-resize",
          "height": "20px",
          "position": "absolute",
          "right": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-10px",
          "cursor": "sw-resize",
          "height": "20px",
          "left": "-10px",
          "position": "absolute",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "nw-resize",
          "height": "20px",
          "left": "-10px",
          "position": "absolute",
          "top": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
  </span>
</div>
`;
